src/Plugins/SwapAnimation/SwapAnimation.js

```diff
@@ -96,25 +98,31 @@ export default class SwapAnimation extends AbstractPlugin {
 
 /**
  * Animates two elements
- * @param {HTMLElement} top
- * @param {HTMLElement} bottom
+ * @param {HTMLElement} from
+ * @param {HTMLElement} to
  * @param {Object} options
  * @param {Number} options.duration
  * @param {String} options.easingFunction
+ * @param {String} options.horizontal
  * @private
  */
-function animate(top, bottom, {duration, easingFunction}) {
-  const height = top.offsetHeight;
+function animate(from, to, {duration, easingFunction, horizontal}) {
+  if (horizontal) {
+    const width = from.offsetWidth;
+    from.style.transform = `translate3d(${width}px, 0, 0)`;
+    to.style.transform = `translate3d(-${width}px, 0, 0)`;
+  } else {
+    const height = from.offsetHeight;
+    from.style.transform = `translate3d(0, ${height}px, 0)`;
+    to.style.transform = `translate3d(0, -${height}px, 0)`;
+  }
 
-  for (const element of [top, bottom]) {
+  for (const element of [from, to]) {
     element.style.pointerEvents = 'none';
   }
```

Comment:
We may want to assign `pointerEvents = 'none';` first (before applying the transforms)